test/08_api-keys.ts

```diff
@@ -107,6 +111,73 @@ describe('create provisioning apikey', function () {
 				);
 			});
 
+			it('should be able to create a provisioning key with a valid expiry date', async function () {
+				const applicationId = this.application.id;
+				const tomorrowDate = new Date(Date.now() + 86400000); // one day in future
+				const { body: provisioningKey } = await fn(
+					this.user,
+					applicationId,
+					`provision-key-${applicationId}-with-expiry`,
+					`Sample key for application-${applicationId} description.`,
+					tomorrowDate,
+				).expect(200);
+
+				expect(provisioningKey).to.be.a('string');
+				this.provisioningKey = provisioningKey;
+
+				// check the name assigned
+				const apiKeyResp = await api.resin.get({
+					resource: 'api_key',
+					passthrough: {
+						req: permissions.root,
+					},
+					id: {
+						key: provisioningKey,
+					},
+					options: {
+						$select: ['name', 'description', 'expiry_date'],
+					},
+				});
+
+				expect(apiKeyResp).to.have.property(
+					'name',
+					`provision-key-${applicationId}-with-expiry`,
+				);
+				expect(apiKeyResp).to.have.property(
+					'description',
+					`Sample key for application-${applicationId} description.`,
+				);
+
+				expect(apiKeyResp).to.have.property('expiry_date');
+				expect(apiKeyResp!.expiry_date.getTime()).to.equal(
+					tomorrowDate.getTime(),
+				);
+			});
+
+			it('should not be able to register a device with an expired provisioning key', async function () {
+				const applicationId = this.application.id;
+				const yesterdayDate = new Date(Date.now() - 86400000); // one day in past
+				const { body: provisioningKey } = await fn(
+					this.user,
+					applicationId,
+					`provision-key-${applicationId}-expired`,
+					`Sample key for application-${applicationId} description.`,
+					yesterdayDate,
+				).expect(200);
+
+				expect(provisioningKey).to.be.a('string');
+
+				await supertest()
+					.post(`/device/register?apikey=${provisioningKey}`)
+					.send({
+						user: this.user.id,
+						application: this.application.id,
+						device_type: 'raspberry-pi',
+						uuid,
+					})
+					.expect(401);
+			});
+
```

Comment:
Can we also add a test case about changing the expiry date w/ a PATCH?
(since that will work in oB b/c there are no whitelists)